### ISSUE TITLE
Crash when enabling debug in Constructron-Continued

### ISSUE THREAD

**Issue by h7896v7347** [JensForstmann/BlueprintSignals#2]
`/ctron enable debug` is causing crash 
![obraz_2023-06-20_130356524](https://github.com/JensForstmann/BlueprintSignals/assets/1020390/51a1c8f0-b9ff-46c0-bea9-af3d3f04eaf4)

https://mods.factorio.com/mod/Constructron-Continued

**Comment by kryojenik** [comment]
Got the same issue, however triggered by LTN.   Any time a global-runtime-setting is changed via script (i.e. not using the mod-settings GUI) this will happen in BlueprintSignals_continued.

8657.040 Error MainLoop.cpp:1382: Exception at tick 17589266: The mod LTN - Logistic Train Network (1.18.3) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event LogisticTrainNetwork::ltn-toggle-dispatcher (ID 245)
The mod Blueprint Signals (continued) (0.4.2) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event BlueprintSignals_continued::on_runtime_mod_setting_changed (ID 61)
__BlueprintSignals_continued__/control.lua:103: bad argument #3 of 3 to '__index' (string expected, got nil)
stack traceback:
	[C]: in function '__index'
	__BlueprintSignals_continued__/control.lua:103: in function <__BlueprintSignals_continued__/control.lua:102>
stack traceback:
	[C]: in function '__newindex'
	__LogisticTrainNetwork__/script/hotkey-events.lua:13: in function <__LogisticTrainNetwork__/script/hotkey-events.lua:8>
8657.040 Error ServerMultiplayerManager.cpp:92: MultiplayerManager failed: "The mod LTN - Logistic Train Network (1.18.3) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event LogisticTrainNetwork::ltn-toggle-dispatcher (ID 245)
The mod Blueprint Signals (continued) (0.4.2) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event BlueprintSignals_continued::on_runtime_mod_setting_changed (ID 61)
__BlueprintSignals_continued__/control.lua:103: bad argument #3 of 3 to '__index' (string expected, got nil)
stack traceback:
	[C]: in function '__index'
	__BlueprintSignals_continued__/control.lua:103: in function <__BlueprintSignals_continued__/control.lua:102>
stack traceback:
	[C]: in function '__newindex'
	__LogisticTrainNetwork__/script/hotkey-events.lua:13: in function <__LogisticTrainNetwork__/script/hotkey-events.lua:8>"

https://github.com/JensForstmann/BlueprintSignals/blob/4e946d717e2b90327b64fef535f43d60a6487f1e/control.lua#L100-L105

on_runtime_mod_setting_changed.player_index can be nil
https://lua-api.factorio.com/latest/events.html#on_runtime_mod_setting_changed

This function should probably be something like this, assuming it will still resolve the crash scenario that caused you to add it in the first place.   I did not explore that.

``` lua
add_event_handler(
        defines.events.on_runtime_mod_setting_changed,
        function(event)
            if not event.player_index then
                return
            end

            GUI.setup(game.players[event.player_index])
        end
)
```

**Comment by JensForstmann** [comment]
Fixed (a long time ago) 9d3ff4fcfea8dce2749d3a26ae1ad09f40406b99
